src/pages/[locale]/products.tsx

```diff
@@ -0,0 +1,182 @@
+import get from 'lodash.get';
+import { IncomingMessage } from 'http';
+
+import Layout from '../../components/Layout';
+import Products from '../../components/Products';
+import Text from '../../components/Text';
+import { PRODUCTS_PAGE_QUERY } from '../../gql/pages/products';
+import withApollo from '../../hoc/withApollo';
+import { Cookies } from '../../types/Cookies';
+import { Customer } from '../../types/data/Customer';
+import { Product } from '../../types/data/Product';
+import { UniversalSSRPageProps } from '../../types/UniversalSSRPageProps';
+import { UserSemiPersistentSession } from '../../types/UserSemiPersistentSession';
+import { prepareGraphCMSLocaleHeader } from '../../utils/graphcms';
+import { DEFAULT_LOCALE, resolveFallbackLanguage } from '../../utils/i18n';
+import { fetchTranslations, I18nextResources } from '../../utils/i18nextLocize';
+import UniversalCookiesManager from '../../utils/UniversalCookiesManager';
+
+const fileLabel = 'pages/products';
+const logger = createLogger({ // eslint-disable-line no-unused-vars,@typescript-eslint/no-unused-vars
+  label: fileLabel,
+});
+
+type Props = {
+  products: Product[];
+} & UniversalSSRPageProps;
+
+const ProductsPage: NextPage<Props> = (props): JSX.Element => {
+  const { products } = props;
+  const productsPublished = filter(products, { status: 'PUBLISHED' });
+  const productsDraft = filter(products, { status: 'DRAFT' });
+
+  Sentry.addBreadcrumb({ // See https://docs.sentry.io/enriching-error-data/breadcrumbs
+    category: fileLabel,
+    message: `Rendering ${fileLabel}`,
+    level: Sentry.Severity.Debug,
+  });
+
+  return (
+    <Layout
+      pageName={'products'}
+      headProps={{
+        title: `${size(products)} products (SSR) - Next Right Now`,
+      }}
+      {...props}
+    >
+      <Container
+        className={'container-white'}
+      >
+        <h1>Products</h1>
+
+        <Text>
+          {`
+            This page uses server side rendering (SSR)
+
+            Each page refresh (either SSR or CSR) queries the GraphQL API and displays products below:
+          `}
+        </Text>
+
+        <hr />
+
+        <h2>Published products</h2>
+
+        <Products
+          products={productsPublished}
+        />
+
+        <hr />
+
+        <h2>Draft products</h2>
+
+        <Text>
+          {`
+            Those products are being created/updated by the NRN community, anybody can manipulate those through <a href="https://nrn-admin.now.sh/#/Product/create" target="_blank">the Admin site</a>.
+
+            Don't hesitate to give it a try, you'll see the list of products below will update because content is fetched for every page request.
+          `}
+        </Text>
+
+        <Products
+          products={productsDraft}
+        />
+      </Container>
+    </Layout>
+  );
+};
+
+ProductsPage.getInitialProps = async (context: NextPageContext & { apolloClient: ApolloClient<NormalizedCacheObject> }): Promise<Props> => {
```

Comment:
It might be better to utilize `getServerSideProps` here, and I feel that there too many things happening here, maybe extract it out into an HOC will be better.

Reply:
I just did! Didn't see your comment before though, I just notices the /products page was failing during CSR, because I hadn't check for req/res availability. I refactored it using getServerSideProps and it looks less tricky. I also refactored some SSR common stuff.

It's not yet perfect, lots of duplicated boilerplate code for GQL queries, but slowly getting there.

Reply:
Oh, amazing! :)